src/table/boot.rs

```diff
@@ -531,12 +531,10 @@ impl BootServices {
     }
 
     /// Locates the handle to a device on the device path that supports the specified protocol.
```

Comment:
Docstring should describe how the `device_path` will be updated. I'm not 100% clear on the behavior from looking at the spec, but maybe something like this?

```rust
/// Locates the handle to a device on the device path that supports the specified protocol.
///
/// The `device_path` is updated to point at the remaining part of the [`DevicePath`] after
/// the part that matched the protocol. If the first node of `device_path` matches the
/// protocol, the `device_path` is advanced to the device path terminator node. If `device_path`
/// is a multi-instance device path, the function will operate on the first instance.
```